patches/components-translate-core-browser-translate_manager.cc.patch

```diff
@@ -0,0 +1,94 @@
+diff --git a/components/translate/core/browser/translate_manager.cc b/components/translate/core/browser/translate_manager.cc
+index a86a2f099b9188a9037f3fd8fff34dfcb1cf451f..392a772bef8fb8cdc827488eacce105fa921da0d 100644
+--- a/components/translate/core/browser/translate_manager.cc
++++ b/components/translate/core/browser/translate_manager.cc
+@@ -414,7 +414,7 @@ void TranslateManager::TranslatePage(const std::string& original_source_lang,
+   // kUnknownLanguageCode in order to send a translation request with enabling
+   // server side auto language detection.
+   std::string source_lang(original_source_lang);
+-  if (!TranslateDownloadManager::IsSupportedLanguage(source_lang))
```

Comment:
apart from the major problem with the number of lines we are changing in this patch (all of which could be replaced by  chromium_src override `#define IsSupportedLanguage IsSupportedSourceLanguage`, this doesn't cover all usages of `TranslateDownloadManager::IsSupportedLanguage` so TranslateManager is going to be out of sync with TranslateInfobarDelegate, TranslateBridge and any future calls to this method that we won't even know about

Reply:
That's wrong. We can't simply use `#define IsSupportedLanguage IsSupportedSourceLanguage`. Previously `IsSupportedLanguage` was used to find out if the language is supported. Now we have asymmetric source and target language lists, so we add two more methods called `IsSupportedSourceLanguage` and `IsSupportedTargetLanguage`, while `IsSupportedLanguage` still exists and returns if the language is contained in at least one of these lists.